test/inductor/test_codecache.py

```diff
@@ -152,6 +155,121 @@ def test_cache_load_model(self, device, dtype):
         self.assertEqual(counters["inductor"]["fxgraph_cache_miss"], 1)
         self.assertEqual(counters["inductor"]["fxgraph_cache_hit"], 1)
 
+    @largeTensorTest("9GB", device="cuda")
+    @config.patch({"fx_graph_cache": True})
+    @parametrize("device", ("cuda",))
+    @parametrize("dtype", (torch.float32, torch.bfloat16))
+    def test_cache_load_with_guards_int32_bounds(self, device, dtype):
+        """
+        Test caching the same graph, but under conditions that introduce guards
+        for tensor sizes < int32.
+        """
+        if device == "cuda" and not HAS_CUDA:
+            raise unittest.SkipTest("requires CUDA")
+        if device == "cuda" and dtype == torch.bfloat16 and not SM80OrLater:
+            raise unittest.SkipTest("requires SM80 or later")
+
+        def fn(x, y):
+            return (x + x, y + y)
+
+        compiled_fn = torch.compile(fn)
+
+        # Mark all tensor arg dimensions as dynamic to cause all shapes
+        # to be symbolic
+        def rand_dynamic(*args):
+            t = torch.rand(*args, device=device, dtype=dtype)
+            for i in range(len(args)):
+                torch._dynamo.mark_dynamic(t, i)
+            return t
+
+        # 1) Both args' shapes are small. We expect guards testing that
+        # both do not exceed int32.
+        a = rand_dynamic(5, 5)
+        b = rand_dynamic(7, 7)
+
+        torch._dynamo.reset()
+        self.assertEqual(fn(a, b), compiled_fn(a, b))
+        self.assertEqual(counters["inductor"]["fxgraph_cache_miss"], 1)
+        self.assertEqual(counters["inductor"]["fxgraph_cache_hit"], 0)
+
+        # A second call should hit.
+        torch._dynamo.reset()
```

Comment:
do we have an API for resetting the on disk compiled graph cache?

Reply:
We don't, but I could add one. Do you have a use case in mind where we'd want to wire it up, or you just want to make it available? There are other caches in this file and the clear() methods don't remove on-disk artifacts. Would we want to add those? Would we want to an API to just blow away all artifacts (independent of type)?

Reply:
The testing API probably looks a bit different; instead of blowing away the full cache, you probably just want to cache into some hermetic directory separate from everything else.